gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/distributed/message/LeadNodeExecutorMsg.java

```diff
@@ -128,6 +128,16 @@ protected void execute() throws Exception {
                 "LeadNodeExecutorMsg.execute: Sent Last result ");
       }
     } catch (Exception ex) {
+
+      Throwable cause = ex ;
+      while(cause != null) {
+        if (cause.getClass().getName().contains("ErrorLimitExceededException")) {
+          throw StandardException.newException(
```

Comment:
I think we can have the while loop for all the exception. See the cause and decide what to throw. If any subtype needs special handling then we can have a nested while loop.